handlers/put.js

```diff
@@ -11,10 +11,8 @@ module.exports = (route, options) => {
     const condition = hu.getIdQuery(options, req);
     req.payload[options.idKey] = hu.getId(options, req);
 
-    const query = Model.updateOne(condition, req.payload, {
-      overwrite: options.overwrite,
-      upsert: options.upsert,
-    }).lean();
+    const { overwrite, upsert } = options;
+    const query = Model.update(condition, req.payload, { overwrite, upsert }).lean();
```

Comment:
I can see why you have a problem changing from `update()` -> `updateOne()` here. If you want to allow `overwrite` you should change this to:

```
const { overwrite, upsert } = options;
const query = overwrite ? Model.replaceOne(condition, req.payload, { upsert }) : Model.updateOne(condition, req.payload, { upsert });
```

Here are the [relevant mongoose docs](https://mongoosejs.com/docs/deprecations.html#-update-).

Sorry for the inconvenience here, this was a change in the MongoDB server so Mongoose unfortunately can't make `updateOne()` 100% compatible with how `overwrite` works. Specifically, if you specify `overwrite: true` you can still pass atomic operators to Mongoose's `Model.update()`, but you can't pass atomic operators to `Model.replaceOne()`

Reply:
@vkarpov15  thank you so very much for also participating here!

The approach you posted above was my first try on this issue. I ran into some "empty sub-documents and arrays with only `null` items" issue (see above), but this might have had something to do with the mongoDB version 3.6 or an older mongoose version I used at this time.

I've already had a talk with @fiws about it: I will go back to using the switch between Mongoose's `updateOne()` and `replaceOne()` just the way you showed above and test it again. This might be the cleanest solution for this use case (updating/overwriting one document).

Reply:
thanks. looks good! :)